test/IntegrationTests/TicketService/test_delete_custom_field.py

```diff
@@ -0,0 +1,219 @@
+                          f"Custom field {field['field_name']} not found in initial list")
+
+        # Step 3: Delete one custom field (let's delete the second one in the list)
+        field_to_delete = custom_fields[1]
+        url = f"{self.BASE_URL}/custom_fields/{field_to_delete['field_name']}?customer_guid={self.valid_customer_guid}"
+        logger.info(f"Deleting custom field: {field_to_delete['field_name']}")
+
+        # Perform the deletion
+        response = requests.delete(url)
+        self.assertEqual(response.status_code, HTTPStatus.OK,
+                         f"Failed to delete custom field {field_to_delete['field_name']}")
+
+        # Validate the deletion response
+        response_data = response.json()
+        self.assertEqual(response_data["field_name"], field_to_delete["field_name"])
+        self.assertEqual(response_data["status"], "deleted")
+
+        # Step 4: Verify the remaining 2 custom fields are present
+        updated_fields = self.list_custom_fields()
+        remaining_field_names = [f["field_name"] for f in updated_fields]
+
+        # Verify that the deleted field is no longer present
+        self.assertNotIn(field_to_delete["field_name"], remaining_field_names,
+                         f"Deleted field {field_to_delete['field_name']} still present")
+
+        # Verify that the other 2 fields are still present
+        for field in custom_fields:
+            if field["field_name"] != field_to_delete["field_name"]:
+                self.assertIn(field["field_name"], remaining_field_names,
+                              f"Remaining field {field['field_name']} not found after deletion")
+
+        logger.info("Successfully added 3 custom fields, deleted one, and verified remaining fields.")
+
+    def test_invalid_custom_field_endpoint(self):
+        """Test that an invalid custom field API endpoint returns 404 Not Found."""
+
+        # Define the invalid API endpoint URL
+        invalid_url = f"{self.BASE_URL}/custom_fields/test_field/blah/blah/blah?customer_guid={self.valid_customer_guid}"
+
+        logger.info(f"Testing invalid API endpoint: {invalid_url}")
+
+        # Send a GET request to the invalid URL
+        response = requests.get(invalid_url)
+
+        # Assert that the response status code is 404 Not Found
+        self.assertEqual(response.status_code, HTTPStatus.NOT_FOUND,
+                         f"Expected 404 Not Found, but got {response.status_code} for URL: {invalid_url}")
+
+        # Optionally, check that the response contains a message indicating the endpoint does not exist
+        response_data = response.json()
+        self.assertIn("detail", response_data, "Response should contain a 'detail' key.")
+        self.assertEqual(response_data["detail"], "Not Found", "Expected 'Not Found' message in the response.")
+
+        logger.info(f"Successfully validated 404 Not Found response for invalid endpoint: {invalid_url}")
+
+    def tearDown(self):
```

Comment:
You can also delete the customer that you have created in setUp?

Reply:
sir, we have delete customer db method in database manager, but we dont have a api end point, can i create new one.